### PR TITLE
feat: add post conditions for swapping

### PR DESCRIPTION
**Pull request by philipdesmedt**

### CODE REVIEW

**Comment by philipdesmedt** [comment]
Should be ready, will try some swaps on mainnet with this
image

**Comment by philipdesmedt** [comment]
This doesn't work yet
https://explorer.stacks.co/txid/0xc0c2112c40bff5e25abe847469dc3125ab5f8b77c64fcd077c07f67ff7453fa8?chain=mainnet

**Comment by nieldeckx** [comment]
Still not working in prod :( Even with 100% slippage.

https://explorer.stacks.co/txid/0xd76357d3f538472a58434c6b4f8d47ddd3d8dc52fd10c3251ea5fddf6fb8133f?chain=mainnet
https://explorer.stacks.co/txid/0x3fa259de8e71284ac1f2bcc907d3681637a40dd984b1a168d239b93e92fa03b6?chain=mainnet
https://explorer.stacks.co/txid/0xbb2f72b6725dd0ff037787705655d80c444eb572ff6466f286437afdb9458dfa?chain=mainnet

**Comment by philipdesmedt** [comment]
@nieldeckx maybe try removing the post condition on wSTX altogether? This might not be important

File: web/common/vault-utils.ts [hunk 1]
```diff
@@ -301,3 +310,81 @@ export const contractsMap = {
...
+    makeStandardFungiblePostCondition(
+      sender,
+      FungibleConditionCode.Equal,
+      amountSent,
+      createAssetInfo(tokenX['address'], tokenX['fullName'], tokenX['nameInPair'])
+    )
+  )
+
+  if (tokenZ != undefined) {
+    console.log("GOT Z:", tokenZ);
```
Comment (philipdesmedt) [inline web/common/vault-utils.ts:332]: REQ: should we clean this up?

**Comment by philipdesmedt** [comment]
@nieldeckx I had this running in production but got a report from a user that a WELS/STX transaction failed. I need to check if it is true (didn't ask them for a TX) but I rolled this back in the meantime.

**Comment by nieldeckx** [comment]
@philipdesmedt had a look yesterday and the WELSH swaps failed indeed. Also noticed xBTC swaps failing. No idea why :P

https://explorer.stacks.co/txid/0x99561e428c130572800deefa4703b899439fef245a1ed6bbf9c5aa8f2199ff4a?chain=mainnet
https://explorer.stacks.co/txid/0x2b019bd86cb1d04d824251146245ab9c78c73005aa2d69cf0081bc29a7898838?chain=mainnet

https://explorer.stacks.co/txid/0x3e43a1f632d7afa5757a99fa5025c11e40e598a7afabb90b0d3b2dd36ae041fc?chain=mainnet
https://explorer.stacks.co/txid/0xea7d26a495b068f68ce64b6444c24836f259c10ccca8894077a00315fea7eb48?chain=mainnet

**Comment by nieldeckx** [comment]
@philipdesmedt fixed the welsh and xbtc issue (was using wrong name in pc)

**Comment by philipdesmedt** [comment]
@nieldeckx great - think this should work now. gonna merge

**Pull request merged**